src/main/java/ru/test/restservice/api/Oauth2ApiController.java

```diff
@@ -0,0 +1,94 @@
+package ru.test.restservice.api;
+
+import com.owlike.genson.Genson;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import ru.test.restservice.dao.UserRepository;
+import ru.test.restservice.entity.User;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Map;
+
+@RestController
+@RequiredArgsConstructor
+public class Oauth2ApiController {
+
+    private final UserRepository userRepository;
+
+    @Value("${spring.security.oauth2.client.registration.spbstu.client-id}")
+    private String client_id;
+    @Value("${spring.security.oauth2.client.registration.spbstu.client-secret}")
+    private String client_secret;
```

Comment:
можно короче создать отдельный файл:
```
@Data 
@ConfigurationProperties("spring.security.oauth2.client.registration.spbstu")
public class OAuthProperties {
    private String clientId;
    private String clientSecret;
}
```
и просто инжектить это сюда
```suggestion
final OAuthProperties oauthProperties;
```

Reply:
Этот бин будет только здесь использоваться. Смысл от него тогда